backbone.js

```diff
@@ -1511,13 +1535,18 @@
 
     // Convert a route string into a regular expression, suitable for matching
     // against the current location hash.
-    _routeToRegExp: function(route) {
+    _routeToRegExp: function(route, paramNames) {
```

Comment:
I was thinking about this too. What if this method, instead of returning RegExp, returns an object `{routeRegExp, paramNames}`. It would keep it functionally pure.